### PR TITLE
Functional Prototype

### PR DESCRIPTION
**Pull request by connerbrooks**
Functional prototype running in the kitchen.
Allows user to:
:coffee: badge in
:coffee: create user
:coffee: grind beans
:coffee: track coffee usage
:coffee: get credit for bringing in coffee based materials

### CODE REVIEW

File: CoffeeAuth/DrinkerDatabase.cs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CoffeeAuth
+{
+    class DrinkerDatabase
+    {
+        private static DrinkerDatabase database;
+        public SQLiteConnection conn;
```
Comment (ooeygui) [inline CoffeeAuth/DrinkerDatabase.cs:15]: Why not an Azure table?

File: CoffeeAuth/MainPage.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        protected override void OnNavigatedTo(NavigationEventArgs e)
+        {
+            base.OnNavigatedTo(e);
+            var users = DrinkerDatabase.Instance.GetAllUsers();
+            listView.ItemsSource = users;
+        }
+
+        private void coffee_Click(object sender, RoutedEventArgs e)
+        {
+            if (badgeCIN.Text.Length != 0)
```
Comment (ooeygui) [inline CoffeeAuth/MainPage.xaml.cs]: if statements - even single line - should be wrapped in braces to reduce future programming errors.

File: CoffeeAuth/MainPage.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                this.Frame.Navigate(typeof(UserPage), badgeCIN.Text);
+        }
+        
+
+        private async void badgeCIN_LostFocus(object sender, RoutedEventArgs e)
+        {
+            await Task.Delay(1000);
```
Comment (ooeygui) [inline CoffeeAuth/MainPage.xaml.cs:88]: Why the wait?

Comment (ooeygui) [inline CoffeeAuth/MainPage.xaml.cs:88]: Magic numbers should be constants with an appropriate name

File: CoffeeAuth/Models/User.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        public long NumBags;
+        public long NumMilks;
+        public long NumShots;
+        public long NumLogins;
+
+        public string PictureUrl;
+
+        public override string ToString()
+        {
+            return Name + "\t\t" + Balance + "$";
```
Comment (ooeygui) [inline CoffeeAuth/Models/User.cs]: Localize the currency symbol

File: CoffeeAuth/Package.appxmanifest [hunk 5]
```diff
@@ -0,0 +1,31 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Package xmlns="http://schemas.microsoft.com/appx/manifest/foundation/windows10" xmlns:mp="http://schemas.microsoft.com/appx/2014/phone/manifest" xmlns:uap="http://schemas.microsoft.com/appx/manifest/uap/windows10" IgnorableNamespaces="uap mp">
+  <Identity Name="21b45950-0f39-49cd-bab9-25b119d49918" Publisher="CN=t-cobroo" Version="1.0.7.0" />
+  <mp:PhoneIdentity PhoneProductId="21b45950-0f39-49cd-bab9-25b119d49918" PhonePublisherId="00000000-0000-0000-0000-000000000000" />
+  <Properties>
+    <DisplayName>CoffeeAuth</DisplayName>
+    <PublisherDisplayName>t-cobroo</PublisherDisplayName>
```
Comment (ooeygui) [inline CoffeeAuth/Package.appxmanifest]: Microsoft

File: CoffeeAuth/Properties/AssemblyInfo.cs [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// General Information about an assembly is controlled through the following 
+// set of attributes. Change these attribute values to modify the information
+// associated with an assembly.
+[assembly: AssemblyTitle("CoffeeAuth")]
+[assembly: AssemblyDescription("")]
+[assembly: AssemblyConfiguration("")]
+[assembly: AssemblyCompany("")]
```
Comment (ooeygui) [inline CoffeeAuth/Properties/AssemblyInfo.cs]: Microsoft

File: CoffeeAuth/UserPage.xaml.cs [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+        }
+
+        protected override void OnNavigatedTo(NavigationEventArgs e)
+        {
+            base.OnNavigatedTo(e);
+            badgeCIN = e.Parameter as string;
+
+
+            // check if user is in the database
+            m_user = DrinkerDatabase.Instance.GetUser(badgeCIN);
```
Comment (ooeygui) [inline CoffeeAuth/UserPage.xaml.cs]: Style: m_ shouldn't be used in C#

File: CoffeeAuth/UserPage.xaml.cs [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+            }
+        }
+
+        DispatcherTimer timer;
+        ContentDialog countdown_dialog;
+
+
+        private void getCoffeeButton_Click(object sender, RoutedEventArgs e)
+        {
+            showToast("Espresso Shot", "$1 debited from your account", "thank you for shopping!");
```
Comment (ooeygui) [inline CoffeeAuth/UserPage.xaml.cs]: amount should be configurable; text should be localizable

File: CoffeeAuth/UserPage.xaml.cs [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+        }
+
+
+        private async void Countdown()
+        {
+            TextBlock text = new TextBlock();
+            text.HorizontalAlignment = HorizontalAlignment.Center;
+            text.Text = "You have 30 seconds to grind your coffee";
+
+            countdown = new TextBlock();
```
Comment (ooeygui) [inline CoffeeAuth/UserPage.xaml.cs]: UI Block should be in XAML not manually constructed.

File: CoffeeAuth/UserPage.xaml.cs [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+
+        private void AppBarButton_Click(object sender, RoutedEventArgs e)
+        {
+            Frame.Navigate(typeof(MainPage));
+        }
+
+        private void showToast(string heading, string body, string body2)
+
+        {
+            var builder = new StringBuilder();
```
Comment (ooeygui) [inline CoffeeAuth/UserPage.xaml.cs]: This looks hinky. Can you load the XAML instead of constructing it from a string?

**Comment by ooeygui** [comment]
Add appropriate license file

File: CoffeeAuth/MainPage.xaml.cs (hunk 3 above)
Comment (connerbrooks) [inline CoffeeAuth/MainPage.xaml.cs:88]: If you immediately refocus after loosing focus you cannot interact with the application. I need to keep focus on the textbox for badge input.

Comment (ooeygui) [inline CoffeeAuth/MainPage.xaml.cs:88]: Is that a system bug you are working around? Or is this an application problem?

**Comment by ooeygui** [comment]
Could you add the Microsoft copyright headers with the MIT License?

**Pull request merged**